### DISCUSSION TITLE
Importing custom configs/recipes in YAML

### DISCUSSION THREAD

**Discussion by SimJoSt** [deployphp/deployer#3653] [Help needed]
We really like the simplicity of the YAML syntax to create most of the configs. To create some custom configs and recipes, we have the following file structure:

- deploy.yaml
- .deployer/
    - hosts.yaml
    - config.yaml
    - custom.php

How can you import the files in the `.deployer` directory from the main `deploy.yaml` config file?
The `- import: recipe/common.php` statements seem to always use the path of the deployer directory in the composer vendors.
Is there a way to use import differently, to import files relative to the main config file? Like in PHP with `__DIR__`, etc.?

**Comment by antonmedv** [comment]
> The - import: recipe/common.php statements seem to always use the path of the deployer directory in the composer vendors.
Is there a way to use import differently, to import files relative to the main config file? Like in PHP with __DIR__, etc.?

I think we need to update the code

     require $path;

And join with base recipe path by default.

**Reply by antonmedv** [reply]
Create a tracking issue https://github.com/deployphp/deployer/issues/3655
